.gitignore

```
@@ -1,26 +1,31 @@
-*.DS_Store
-.project
+!*.gitkeep
+
+.nyc_output/*
+# .vscode/settings.json
 node_modules/*
+
+analyzer/
+bower-repo/
+coverage/
+examples/3d/
 experiments/*
+dist/
+docs/reference/*
+docs/data.json
 lib_old/*
 lib/p5.*
+lib/p5-test.js
 lib/modules
-docs/reference/*
-!*.gitkeep
-examples/3d/
-.idea
-dist/
-*.d.ts
-p5.zip
-bower-repo/
 p5-website/
-.vscode/settings.json
-.nyc_output/*
-coverage/
-lib/p5-test.js
+# preview/

```

```suggestion
preview/
```

This is meant to be ignored. It isn't quite the perfect setup yet but the idea is that people can use this folder as they wish to test out their implementation directly. I can't find an ideal solution to ignore changes to tracked files in git which is the main thing it can't quite achieve yet, but anyone can still add files into this folder for testing purposes and it not be committed.

---

This way the files in `preview/` won't be linted. But this should be tackled in a follow-up. Linting the source files has a higher priority